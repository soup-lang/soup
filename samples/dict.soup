func first(x, y)
    return x

func second(x, y)
    return y

func sum(x, y)
    return x + y

x: {
    'first': first,
    'second': second,
    'sum': sum
}

console.log(x['sum'](1, 4))
console.log(x['first'](1, 4))
console.log(x['second'](1, 4))

console.log({
    first: first,
    second: second,
    1: sum
}[1]('shinto ', 'shrine'))

y: {
    summer: 'xia',
    fall: 'qiu',
    winter: 'dong',
}

y |: console.log($k + ':' + $)

func opMap()
    return {
        '==': '===',
        '!=': '!==',
    }

o: opMap()
console.log(o['=='])
console.log(o['!='])
