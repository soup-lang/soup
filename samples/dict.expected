5
1
4
shinto shrine
summer:xia
fall:qiu
winter:dong
===
!==
